### ISSUE TITLE
UI Presets: Allow users to easily configure Cosmic to look like Windows/KDE/Cinnamon with 1 click

### ISSUE THREAD

**Issue by Pentaphon** [pop-os/cosmic-epoch#746]
One thing that may help users adopt Cosmic is to give users preset desktop configurations like a Windows, KDE or Cinnamon-style desktop. These presets can be added to a future Pop!OS landing screen and the settings page. 

They would get the original COSMIC desktop by default but if they need a more familiar experience or they want to help switch longtime Windows users to COSMIC, they can just easily switch to a preset "Windows mode" so that COSMIC looks like a classic Windows XP desktop.

**Comment by Department-stack** [comment]
i think that a user curated theme store would be better. then if someone would want to make a kde-like desktop they could maintain it themselves and upload it for others to use. i personally didn't need to adapt when switching from kde to cosmic. linux newbs could always just use kde if its more familar to them. seems like a lot of work for a very few minority but is an interesting proposal nonetheless

**Comment by Pentaphon** [comment]
> i think that a user curated theme store would be better

I understand why you would want this but this would simply take forever for users to set up, especially since COSMIC is so new. I think it would be better if the devs just added 1 official Windows-style UI config in the settings as a preset so that it is ready for users to use on Day 1 of COSMIC going stable. A user-curated store can come much later for people that want to do more advanced theming.

**Comment by 9SMTM6** [comment]
> Perhaps a user can submit a PR to help with this if enough people want such a configuration preset.

I don't think enough people will want to, and I don't think asking this to be 

> ready for users to use on Day 1 of COSMIC going stable

is in any way shape or form an acceptable ask.

I honestly would be surprised if the cosmic devs would even accept a perfect PR that does this, at least in the current situation. Even if it works perfectly right now in every situation, they would have to test for this again and again after changes were made.

Such a thing is a nightmare for quality assurance.

The most I'd expect is them opening the doors for user generated modifications, and I'd not expect that API to be anything like complete for some time after the stable release.

**Comment by Pentaphon** [comment]
> I don't think enough people will want to, and I don't think asking this to be

I think this is more in-demand than you think and many people would welcome a layout of COSMIC that closely matches what MATE, Cinnamon and KDE have done to emulate Windows. A lot of people installing Pop!OS are looking for a Windows alternative, and it would be handy to have a familiar layout available to such users. I don't think every former Windows user will use such a layout but enough of them will give it a shot, especially if they are upgrading somebody else's machine. 

>Such a thing is a nightmare for quality assurance.

It doesn't have to be. It's just a simple layout rearrangement. Some people even use MATE, Cinnamon and KDE for that layout and many people would test such a layout preset if one were made. 

>The most I'd expect is them opening the doors for user generated modifications, and I'd not expect that API to be anything like complete for some time after the stable release.

I'm saying the exact same thing. I'm not asking for heavy modification here. I am just saying that one of the many possible layout rearrangements should be a Window-type layout and it should be easily accessible via the settings page

**Comment by Department-stack** [comment]
"I am just saying that one of the many possible layout rearrangements should be a Windows-type layout and it should be easily accessible via the settings page so that newbies can easily find it and say "oh nice, a Windows-type layout" and click it." 

That seems like a simple user modification to make by yourself then

**Comment by Pentaphon** [comment]
> That seems like a simple user modification to make by yourself then

If I could do it by myself, I'd submit a PR, but I can't, hence my feature request.

**Comment by Department-stack** [comment]
> 
> If I could do it by myself, I'd submit a PR, but I can't, hence my feature request.

I mean in cosmic settings, you could remove the dock, move the panel to the bottom & make it's size bigger

**Comment by Pentaphon** [comment]
> I mean in cosmic settings, you could remove the dock, move the panel to the bottom & make it's size bigger

Yes, but that's still not close to Windows/KDE/Cinnamon desktop. I'll just wait to see if anybody is interested in submitting a PR.

**Comment by Department-stack** [comment]
I think its pretty close (excluding the app launcher) you can remove or move applets to your liking as well

**Comment by Pentaphon** [comment]
> I think its pretty close (excluding the app launcher) you can remove or move applets to your liking as well

The thing that a lot of people like about a Windows-type layout is that everything you need is on a bottom bar so there's no harm in seeing if COSMIC can eventually emulate that behavior for people who want that. If COSMIC is supposed to be customizable, surely there's no harm in providing at least one option that provides that layout. I don't get why anybody would want to get in the way of making this a possibility when there's a Unity-type layout, a Mac-type layout, etc.

**Comment by Pentaphon** [comment]
> I would argue this option should be surfaced during installation/first boot even.

Agreed. It should be in the installer similar to how Ubuntu lets you choose a "minimal installation". A small detail that goes a long way in having people keep Pop!OS after they try it.

**Comment by edfloreshz** [comment]
Tweaks can do pretty much everything you're asking for, hopefully at some point many of the features in Tweaks will be integrated into COSMIC, but for now you can use Tweaks.

https://tweaks.edfloreshz.dev/

**Comment by ilikenwf** [comment]
> One thing that may help users adopt Cosmic is to give users preset desktop configurations like a Windows, KDE or Cinnamon-style desktop. These presets can be added to a future Pop!OS landing screen and the settings page.
> 
> I am not asking for this preset configuration to be themed to look exactly like Windows, but rather, the COSMIC UI in a Windows-type layout with everything at the bottom with the applications button and dock skewed to the left and the notifications on the right.
> 
> This way, new users would get the original COSMIC desktop by default but if they need a more familiar experience or they want to help switch longtime Windows users to COSMIC, they can just easily switch to a preset "Windows mode" so that COSMIC looks like a classic Windows XP desktop.
> 
> In other words, it should be easy for new users to get this look by switching to it via a preset when installing or via the settings page.
> 
> ![one](https://private-user-images.githubusercontent.com/49374058/362026991-c804cf5e-51e4-42af-a090-bb48147ed37c.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3NTkyODQsIm5iZiI6MTczMjc1ODk4NCwicGF0aCI6Ii80OTM3NDA1OC8zNjIwMjY5OTEtYzgwNGNmNWUtNTFlNC00MmFmLWEwOTAtYmI0ODE0N2VkMzdjLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI4VDAxNTYyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhNmQwMmY3YWZhZTNkOTliNGRjMzljNmY1Y2MxNzYzYTNhZWRhNGUwYzFmMWZjMDM0ZjA4ZDJkZjI1OGMyZGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NyzpU98shTG_T9CcSAUqv2M1_jSAr66IKdzeLWVE85I)

So I'm new to COSMIC - did you actually get the rust version configured this way, and if so, you could at least guide me gently as to how?

**Comment by Shadow8472** [comment]
In case anyone ever takes this up, be sure to include default keyboard shortcuts, starting with Ctrl+Alt+T for a terminal.

**Comment by Pentaphon** [comment]
> Tweaks can do pretty much everything you're asking for, hopefully at some point many of the features in Tweaks will be integrated into COSMIC, but for now you can use Tweaks.
> 
> https://tweaks.edfloreshz.dev/

@edfloreshz any updates for tomorrow's beta?
